### PR TITLE
Remove support for inferring the return type of a local function.

### LINKED ISSUE #7832: Rip out support for inferred return type in local functions
The prototype for local functions (in the future branch) currently includes some support for inferring the return type. The LDM decided that we don't want that. Of course it would be possible to relax this in the future, but for the first iteration this is what we want.

Please remove the code and only support explicitly declared return types.

### PR DESCRIPTION
**Pull request by gafter**
Fixes #7832 

@dotnet/roslyn-compiler Please review.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs [hunk 1]
```diff
@@ -220,74 +157,15 @@ internal TypeSymbol ComputeReturnType(BoundBlock body, bool returnNullIfUnknown,
                 // The return type of an async method must be void, Task or Task<T>
                 diagnostics.Add(ErrorCode.ERR_BadAsyncReturn, this.Locations[0]);
             }
+            // TODO: note there is a race condition here that will ultimately need to be fixed.
+            // Specifically, the Interlocked.CompareExchange above succeeds, and will be seen by
+            // other threads, before the diagnostics have been recorded in this symbol, below.
+            // We can resolve this with a lock around this method and any other methods that
+            // manipulate _diagnostics, directly or indirectly.
             AddDiagnostics(diagnostics.ToReadOnlyAndFree());
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs]: Consider logging a bug for TODO.

**Comment by cston** [comment]
LGTM

**Comment by davkean** [comment]
@dotnet-bot retest prtest/win/dbg/unit32 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_win_dbg_unit32/2727/
// Retest reason: build break

File: src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs (hunk 1 above)
Comment (gafter) [inline src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs]: Done (#7938)

**Comment by TyOverby** [comment]
LGTM

**Comment by gafter** [comment]
I'm not sure about links from the compiler code to github. That would be a new precedent. On the other hand, it would make it easier to know which TODOs have already been made into issues.

**Pull request merged**